src/logger.ts

```diff
@@ -139,20 +150,39 @@ export class SilentLogger implements OptionalLogger {}
  *   f(lc2);  // logging inside f will be prefixed with 'foo'
  */
 export class LogContext extends OptionalLoggerImpl {
-  private readonly _logSink: LogSink;
   private readonly _level: LogLevel;
-
-  constructor(level: LogLevel = 'info', logSink: LogSink = consoleLogSink) {
-    super(logSink, level);
+  private readonly _context?: Context;
```

Comment:
Another Roci gotcha. Use:

```ts
  private readonly _context: Context | undefined;
```

and make sure it is initialized in the constructor.

We really want to stay away from optional properties because it leads to polymorphic code (which is slower than monomorphic code).

Reply:
Cool, thanks for the background! I love learning about advanced under-the-hood details of javascript.